components/prize_pool/commons/prize_pool_import.lua

```diff
@@ -0,0 +1,504 @@
+	-- Apply importLimit if set
+	if Import.config.importLimit then
+		local sums = MathUtil.partialSums(Array.map(placementEntries, function(entries) return #entries end))
+		local index = ArrayExt.findIndex(sums, function(sum) return Import.config.importLimit <= sum end)
+		if index ~= 0 then
+			placementEntries = Array.sub(placementEntries, 1, index - 1)
+		end
+	end
+
+	return Import._mergePlacements(placementEntries, inputPlacements)
+end
+
+-- Compute placements and their entries of all brackets or all group tables in a
+-- tournament stage. The placements are ordered from high placement to low.
+function Import._computeStagePlacementEntries(stage, options)
+	local groupPlacementEntries = Array.map(stage, function(matchGroup)
+		return TournamentUtil.isGroupTable(matchGroup)
+			and Import._computeGroupTablePlacementEntries(matchGroup, options)
+			or Import._computeBracketPlacementEntries(matchGroup, options)
+	end)
+
+	local maxPlacementCount = Array.max(Array.map(
+			groupPlacementEntries,
+			function(placementEntries) return #placementEntries end
+		))
+	return Array.map(Array.range(1, maxPlacementCount or 0), function(placementIndex)
+		return Array.flatMap(groupPlacementEntries, function(placementEntries)
+			return placementEntries[placementIndex]
+		end)
+	end)
+end
+
+-- Compute placements and their entries from a GroupTableLeague record.
+function Import._computeGroupTablePlacementEntries(standingRecords, options)
+	local needsLastVs = Import._needsLastVs(standingRecords)
+	local placementEntries = {}
+	local placementIndexes = {}
+
+	for _, record in ipairs(standingRecords) do
+		if options.isFinalStage or Table.includes(options.groupElimStatuses, record.currentstatus) then
+			local entry = {
+				date = record.extradata.endTime and DateExt.toYmdInUtc(record.extradata.endTime),
+				showMatchDraws = record.extradata.showMatchDraws or false,
+			}
+
+			if not record.extradata.placeRange then
+				record.extradata.placeRange = {record.placement, record.placement}
+			end
+			if record.extradata.placeRange[1] == record.extradata.placeRange[2] then
+				Table.mergeInto(entry, {
+					matchScore = record.scoreboard.match,
+					opponent = record.extradata.opponent,
+				})
+				if entry.opponent.template then
+					entry.opponent.name = entry.opponent.template
+				else
+					entry.opponent.isResolved = true
+				end
+			end
+
+			entry.needsLastVs = needsLastVs
+			entry.matches = record.matches
+
+			if not placementIndexes[record.placement] then
+				table.insert(placementEntries, {entry})
+				placementIndexes[record.placement] = #placementEntries
+			else
+				table.insert(placementEntries[placementIndexes[record.placement]], entry)
+			end
+		end
+	end
+
+	return placementEntries
+end
+
+function Import._needsLastVs(standingRecords)
+	if Import.config.allGroupsUseWdl then
+		return false
+	elseif (standingRecords[1] or {}).type == SWISS_GROUP_TYPE then
+		return true
+	elseif #standingRecords ~= GSL_GROUP_OPPONENT_NUMBER then
```

Comment:
Does this work with rounds?

Reply:
it only queries the entries of the last round so yes